### PR TITLE
[FIX] sale_product_configurator: use float quantities in configurator

### PR DESCRIPTION
**Pull request by MerlinGuillaume**
The product configurator doesn't allow you to use non-integer quantities

Steps to reproduce:
1. Install Sales
2. Go to Settings > Sales > Product Catalog and enable Product
   Configurator
3. Go to Sales > Products and create a product 'Product A' with
   attribute Color and at least two values
4. Create a Quotation and add product 'Conference Chair (CONFIG)'
5. In the product configurator, specify 3.5 as quantity and click on
   'ADD'
6. The total price is wrong
7. Add the optional product 'Chair floor protection' and confirm
8. The quantity of 'Chair floor protection' rounds to 3

Solution:
Parse the quantity as a float instead of an integer (and replace
potential commas to dots in case of different decimal separator), get
the quantity from default_quantity for the rpc call (as it was converted
to an integer because of its type) and convert the quantity to float in
the controllers

Problem:
parseInt was used to retrieve the quantity and the controllers converted
the quantity to an integer. The field quantity of the
SaleProductConfigurator is of type Integer, which rounded the quantity
when opening the configurator.

opw-2896367

### CODE REVIEW

**Comment by agr-odoo** [comment]
Weird, the (almost) same changes was presented in https://github.com/odoo/odoo/pull/82438
Discussed, approved, r+'ed... I now realize the PR was closed right before merging, probably by accident @fdamhaut 
I'm ok with the change, the suggested js improvement looks ok as well. As we use the same widget for optional products and variant products configuration we need to ensure it works well for both

File: addons/sale_product_configurator/static/src/js/product_configurator_controller.js
Comment (Feyensv) [inline addons/sale_product_configurator/static/src/js/product_configurator_controller.js:118]: Doesn't this mean that the existing qty will be always kept even if the qty is changed in the wizard ? :thinking:

File: addons/sale_product_configurator/static/src/js/product_configurator_modal.js
Comment (Feyensv) [inline addons/sale_product_configurator/static/src/js/product_configurator_modal.js:182]:
```suggestion
            var quantity = parseFloat($item.find('input[name="add_qty"]').val().replace(',', '.') || 1);
```

File: addons/sale_product_configurator/static/src/js/product_configurator_modal.js
Comment (Feyensv) [inline addons/sale_product_configurator/static/src/js/product_configurator_modal.js:475]:
```suggestion
                var quantity = parseFloat($(this).find('input[name="add_qty"]').first().val().replace(',', '.') || 1);
```

**Review by Feyensv — COMMENTED** [review]
Seems to work fine in the backend (note that for the ecommerce, the amounts are automatically rounded anyway)

**Review by Feyensv — APPROVED** [review]
@robodoo r+

File: addons/sale_product_configurator/static/src/js/product_configurator_modal.js
Comment (Feyensv) [reply addons/sale_product_configurator/static/src/js/product_configurator_modal.js:182]: It doesn't fail, but the qty is rounded to 1.0.
I tried removing the replace part, but parseFloat doesn't like american amounts either, the only change is that the resulting quantity is 1.25 instead of 1.0.

Note that TDE is not responsible for the scope and didn't do anything on this PR. Please avoid tagging people in messages, unless you do not have any answer the standard way (Codeowners, people managing a given PR, ...)

**Pull request closed without merging**
